userguide/tutorials/invoice_subsystem.adoc

```diff
@@ -1199,6 +1199,48 @@ A new node `D`, child of `c`, is inserted representing that period. After the `R
 
 Finally, one last time, the tree is built and the computed invoice items become the resulting items, i.e. the items for the resulting invoice. In our case, two items are generated: the `RECURRING` kept as-is (outside the tree), and a repair from 2012-05-08 to 2012-06-01.
 
+== Keeping invoices in DRAFT mode
+
+In some situations, we would like Kill Bill to generate the invoices, but leave them in a `DRAFT` mode.
+
+Examples:
+
+* Case where we need to append invoice items after generation
+* Case where we want payment system to ignore the invoice until it has been `COMMITTED`
+
+This behavior can be configured at the account level through the `AUTO_INVOICING_DRAFT` tag: when an `Account` is tagged with `AUTO_INVOICING_DRAFT`, any automatic invoice generated by the system (such as when generating invoices for recurring and usage subscriptions) will end up in `DRAFT` mode. Generating invoices through API calls (e.g. adding external chargrs) will not be affected by this tag.
+
+=== Implementation expectation
+
+Any `DRAFT` invoice containing `RECURRING` or `USAGE` items will be taken into consideration by the system when computing subsequent invoices, regardless of the `AUTO_INVOICING_DRAFT` tag.
+
+As an example, let's assume a monthly recurring subscription billed on the 1st of the month:
+
+* `AUTO_INVOICING_DRAFT` is added
+* January 1st : `DRAFT` invoice generated with a `RECURRING` item from January 1st to February 1st
+* `AUTO_INVOICING_DRAFT` is removed
+* February 1st : `COMMITTED` invoice generated with a `RECURRING` item from February 1st to March 1st -- although the previous invoice has not been committed yet
+
+In other words, the feature assumes that at some point such `DRAFT` invoices will be committed by the user. Not doing so will lead to users not being billed for those specific periods.
```

Comment:
I think we are missing out something then, which is the ability to cancel a `DRAFT` invoice. Otherwise, that pretty much means that `DRAFT` invoiced are just delayed. Thoughts?

Reply:
That's part of https://github.com/killbill/killbill/issues/833, no?

Reply:
Ah ah... forgot about that one. Indeed!